src/ranking.jl

```diff
@@ -6,50 +6,63 @@
 # The implementations here follow this wikipedia page.
 #
 
-
 function _check_randparams(rks, x, p)
     n = length(rks)
     length(x) == length(p) == n || raise_dimerror()
     return n
 end
 
+# ranking helper function: calls sortperm(x) and then ranking method f!
+function _rank(f!, x::AbstractArray, R::Type=Int; sortkwargs...)
+    rks = similar(x, R)
+    ord = reshape(sortperm(vec(x); sortkwargs...), size(x))
+    return f!(rks, x, ord)
+end
 
+# ranking helper function for arrays with missing values
+function _rank(f!, x::AbstractArray{>: Missing}, R::Type=Int; sortkwargs...)
+    inds = findall(!ismissing, vec(x))
+    isempty(inds) && return missings(R, size(x))
+    T = nonmissingtype(eltype(x))
+    xv = Vector{T}(undef, length(inds))
+    @inbounds for (i, ind) in enumerate(inds)
+        xv[i] = x[ind]
+    end
+    rks = missings(R, size(x))
+    ordv = sortperm(xv; sortkwargs...)
+    f!(view(rks, inds), xv, ordv)
+    return rks
+end
 
 # Ordinal ranking ("1234 ranking") -- use the literal order resulted from sort
-function ordinalrank!(rks::AbstractArray, x::AbstractArray, p::IntegerArray)
-    n = _check_randparams(rks, x, p)
-
-    if n > 0
-        i = 1
-        while i <= n
-            rks[p[i]] = i
-            i += 1
-        end
+function _ordinalrank!(rks::AbstractArray, x::AbstractArray, p::IntegerArray)
+    _check_randparams(rks, x, p)
+    @inbounds for i in eachindex(p)
+        rks[p[i]] = i
     end
-
     return rks
 end
 
 
 """
-    ordinalrank(x; lt = isless, rev::Bool = false)
+    ordinalrank(x; sortkwargs...)
```

Comment:
Repeating the list of supported arguments is more user-friendly IMO, and it doesn't take a lot of space.

Reply:
I've expanded *lt*, *by* and *rev*, because they are more commonly used, and *alg* and *order* definitions are too long.
The docstring now explicitly refers to the `sort(x)` keywords in the description.